pkg/controller/node/node_controller.go

```diff
@@ -995,3 +1022,92 @@ func getErrorString(err error) string {
 	}
 	return ""
 }
+
+// alertIfImportantFilesPending checks to see if there are any important files in the
+// machineconfig that the pool should be moving to, and alerts if there are
+func (ctrl *Controller) alertIfImportantFilesPending(pool *mcfgv1.MachineConfigPool) error {
+
+	// The config we would be going to
+	spec_name := pool.Spec.Configuration.Name
+	// The config we're in right now
+	status_name := pool.Status.Configuration.Name
+
+	// Get the machine config objects
+	statusConfig, err := ctrl.mcLister.Get(status_name)
+	if errors.IsNotFound(err) {
+		glog.V(2).Infof("MachineConfig %v has been deleted", status_name)
+		return err
+	}
+
+	specConfig, err := ctrl.mcLister.Get(spec_name)
+	if errors.IsNotFound(err) {
+		glog.V(2).Infof("MachineConfigPool %v has been deleted", spec_name)
+		return err
+	}
+
+	// Make sure we can coax the objects into ignitionv3
+	statusIgnConfig, err := ctrlcommon.ParseAndConvertConfig(statusConfig.Spec.Config.Raw)
+	if err != nil {
+		return err
+	}
+	specIgnConfig, err := ctrlcommon.ParseAndConvertConfig(specConfig.Spec.Config.Raw)
+	if err != nil {
+		return err
+	}
+
+	// Go through the files and see what is new or different
+	oldFileSet := make(map[string]ign3types.File)
```

Comment:
Perhaps we can move calculating file diff list logic into another function so that it can be reused here as well as in [calculatePostConfigChangeActionFromFileDiffs() function](https://github.com/openshift/machine-config-operator/blob/master/pkg/daemon/update.go#L424)